mozilla-release/browser/components/sessionstore/SessionStore.jsm

```diff
@@ -2961,19 +2961,44 @@ var SessionStoreInternal = {
     // This is intended to work when both tab restore and fresh tab features
     // are enabled.
     if (!overwriteTabs) {
+      const uriFixup = Services.uriFixup;
+      function fixupURI(pageAddress) {
+        try {
+          return uriFixup.createFixupURI(pageAddress, uriFixup.FIXUP_FLAG_NONE);
+        }
+        catch (e) {
+          return pageAddress;
+        }
+      }
+      function similarURIs(uri1, uri2) {
```

Comment:
User may enter `cliqz.com`, which upon navigation ends up on `https://cliqz.com/`, but `nsIURIFixup`'s protocol guess is `http`. That's why we want an incomplete, non-strict match here. Other possible mismatch to ignore is in-page anchors.